src/theory/quantifiers/sygus/cegis.cpp

```diff
@@ -235,10 +263,20 @@ bool Cegis::constructCandidates(const std::vector<Node>& enums,
     {
       std::vector<Node> fail_cvs = enum_values;
       Assert(candidates.size() == fail_cvs.size());
+      // try to solve entire problem?
       if (src->repairSolution(candidates, fail_cvs, candidate_values))
       {
         return true;
```

Comment:
This PR made me think that it's spurious to do the solution check upstream when we know, in this point here, that `candidate_values` will necessarily work. Maybe we should make `constructCandidates` return an enum rather than a bool to account for fail / candidate / solution?

Reply:
Although it is spurious, I think it's best to re-verify the solution upstream.  This isn't very expensive and makes things more uniform.

Reply:
Yes... I was not too keen on the micro-optimization.